File: Week2/day2/draw-diagonal.py
# Write a program that reads a number from the standard input, then draws a
# square like this:
#
#
# %%%%%
# %%  %
# % % %
# %  %%
# %   %
# %%%%%
#
# The square should have as many lines as the number was
number = int(input('Give me a number, please: '))
if number == 1:
    print('%'*number)
elif number == 2:
    print('%'*number)
    print('%'*number)
elif number != 2 and number != 3:
    print('%'*number)
    for x in range(1, number-1):
        y = x;
        y -=1
        print('%'+' '*(y)+'%'+' '*((number)-y-3)+'%')
    print('%'*number)
elif number == 3:
    print('%'*number)
    print('%%%')
    print('%'*number)

File: Week7/Day2/draw-diagonal.js

'use strict';

var lineCount = 6;


// Write a program that draws a
// square like this:
//
//
// %%%%%
// %%  %
// % % %
// %  %%
// %   %
// %%%%%
//
// The square should have as many lines as lineCount is

var space = '';
var mark = '%%';
var secondSpace;
var numberSpace = lineCount;
var space2 = '';
for (var j = 1; j <= lineCount-2; j++)
{
    mark += '%';
}
console.log(mark);
for (var i = 1; i <= lineCount-2; i++)
{
    space2 = '';
    for (var h = 1; h <= numberSpace-3; h++)
    {
        space2 += ' ';
    }
    console.log('%' + space + '%' + space2+'%');
    numberSpace -= 1;
    space += ' ';
}
console.log(mark);